### PR TITLE
script: aprg - Automatic Performance Report Generator POC toolkit

### PR DESCRIPTION
**Pull request by perezjosibm**
This toolkit is a POC for the execution of performance tests using FIO with RBD, measurement of CPU utilisation for the OSD process (as well as the FIO client), coalescing of data to produce response latency graphs, and generate a report in .pdf
I started this POC before I was aware of CBT,  I've been porting these to CBT from PR 306 and follow ups. 

The following are example of the  types of charts produced by the toolkit:
 
**Response latency curves:**
 
this example compares side by side IOPs, Latency and CPU utilisation Classic vs Crimson OSD: the (vertical) error bars represent the stdev, so the closer they are to the FIO completion latency mean the better the confidence on the measured data point

crimson_1osd_8fio_rc_1procs_randread_iops_vs_lat_vs_cpu_usr

**Single workload execution:**
This  example shows the response latency chart for a Classic OSD configuration, single OSD, system CPU utilisation, 4k random read workload:

classic_1osd_rt_1procs_randread_iops_vs_lat_vs_cpu_sys

Please see the `aprg/README.rst` for further details.
 

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by perezjosibm** [comment]
By mistake, it adds a commit I have for a separate branch and PR, I think I created this branch from that one, hence the problem 😭 
I need to figure out how to indicate that in githhub

File: src/script/aprg/README.rst [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# aprg - Automatic Performance Report Generator - POC toolkit
+
+This is a POC/prototype of a number of ideas to:
+
+- run mechanically a set of workloads using FIO with RBD
```
Comment (anthonyeleven) [inline src/script/aprg/README.rst:5]: s/mechanically/

File: src/script/aprg/README.rst [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+- get some measurements for CPU utilisation, memory
```
Comment (anthonyeleven) [inline src/script/aprg/README.rst:6]: gather measurements for CPU utilization and RAM

Comment (perezjosibm) [reply src/script/aprg/README.rst]: Thank you @anthonyeleven for your comments, I've updated the PR 👍

**Comment by perezjosibm** [comment]
Hi @Matan-B this is the toolkit I've been using during the Crimson performance tracker recommendations, I'll appreciate your review please. Thanks!

**Review by anthonyeleven — COMMENTED** [review]
Docs lgtm

**Comment by perezjosibm** [comment]
jenkins test make check

**Comment by perezjosibm** [comment]
I've created a separate [repo](https://github.com/perezjosibm/ceph-aprg) for this POC toolkit, mostly for convenience to update test plans, etc separately from Ceph, so I will be closing this PR, Cheers.

**Pull request closed without merging**